native/source/mixpanel/detail/worker.cpp

```diff
@@ -135,7 +137,13 @@ namespace mixpanel
             nanowww::Response response;
             if (client.send_request(request, &response))
             {
-                mixpanel->log(Mixpanel::LogEntry::LL_TRACE, "response: " + response.content());
+                // Lock on updating the network requests allowed time
+                {
+                    std::lock_guard<std::mutex> lock(mutex);
+                    // Prevent requests until the back off time has passed
+                    network_requests_allowed_time = parse_www_retry_after(response);
```

Comment:
**NOT A NIT:** Please make the call to `parse_www_retry_after` *outside* of the mutex lock section. Otherwise you're doing a ton of stuff (including memory allocation and calls to `mixpanel->log`) while holding a mutex lock.

Do the computation first, then take the lock and update the variable and (RAII-implicitly) release the lock.